tests/prometheus-custom.sh

```diff
@@ -0,0 +1,78 @@
+# Verifies a custom Prometheus metrics port configuration.
+#
+# @TEST-REQUIRES: $SCRIPTS/docker-requirements
+# @TEST-EXEC: bash %INPUT
+# @TEST-EXEC: btest-diff output.config-staged
+# @TEST-EXEC: btest-diff output.config-deployed
+# @TEST-EXEC: btest-diff output.manager-sd.json
+
+. $SCRIPTS/docker-setup
+. $SCRIPTS/test-helpers
+
+docker_populate singlehost
+
+# The testsuite disables auto-assignment by default.
+
+# Enable Prometheus ports in the created nodes. This technically applies to the
+# agent too, but since it's not running a cluster has no effect there. We could
+# alternatively also specify an extra script to load for every node in the
+# cluster config.
+cat >>zeekscripts/agent-local.zeek <<EOF
+@load frameworks/telemetry/prometheus
+EOF
+
+# Open up additional ports in the all-in-one container, so we can "scrape".
+cat >docker-compose.override.yml <<EOF
+services:
+  controller:
+    expose:
+      - "3000"
+      - "3003"
+EOF
+
+docker_compose_up
+
+# Add curl in the client container. We do this at runtime so it works with both
+# this testsuite's development container and the official Zeek one, as built in
+# our regular CI.
+client_cmd "apt-get -q update && apt-get install -q -y --no-install-recommends curl"
```

Comment:
I'm wondering if at least longer-term we should require curl being pre-installed, and adapt our containers accordingly. Having a test install a system package seems unexpected.

Reply:
I agree, I don't like this either. Adding tooling to the images themselves got controversial last time (see zeek/zeek#3454), but we considered adding a tooling management script in the past. Our CI could run that and commit the resulting image for local testing.

Reply:
I think this is actually causing problems with the tests. From https://api.cirrus-ci.com/v1/task/5518215483228160/logs/test.log:

```
[21:24:17.480] cd testing/external/zeek-testing-cluster && ../../../auxil/btest/btest -A -d -b -j ${ZEEK_CI_BTEST_JOBS}
[21:25:13.200] [#4] tests.prometheus-defaults ... failed
[21:25:13.200]   % 'bash /zeek/testing/external/zeek-testing-cluster/.tmp/tests.prometheus-defaults/prometheus-defaults.sh' failed unexpectedly (exit code 141)
[21:25:13.200]   % cat .stderr
[21:25:13.200]    Network tests_prometheus_defaults__default  Creating
[21:25:13.200]    Network tests_prometheus_defaults__default  Created
[21:25:13.200]    Container tests_prometheus_defaults__client_1  Creating
[21:25:13.200]    Container tests_prometheus_defaults__controller_1  Creating
[21:25:13.200]    Container tests_prometheus_defaults__controller_1  Created
[21:25:13.200]    Container tests_prometheus_defaults__client_1  Created
[21:25:13.200]    Container tests_prometheus_defaults__client_1  Starting
[21:25:13.200]    Container tests_prometheus_defaults__controller_1  Starting
[21:25:13.200]    Container tests_prometheus_defaults__controller_1  Started
[21:25:13.200]    Container tests_prometheus_defaults__client_1  Started
[21:25:13.200]   debconf: delaying package configuration, since apt-utils is not installed
[21:25:13.200] 
[21:25:17.290] [#1] tests.prometheus-multihost ... failed
[21:25:17.290]   % 'bash /zeek/testing/external/zeek-testing-cluster/.tmp/tests.prometheus-multihost/prometheus-multihost.sh' failed unexpectedly (exit code 141)
[21:25:17.290]   % cat .stderr
[21:25:17.290]    Network tests_prometheus_multihost__default  Creating
[21:25:17.290]    Network tests_prometheus_multihost__default  Created
[21:25:17.290]    Container tests_prometheus_multihost__client_1  Creating
[21:25:17.290]    Container tests_prometheus_multihost__controller_1  Creating
[21:25:17.290]    Container tests_prometheus_multihost__inst1_1  Creating
[21:25:17.290]    Container tests_prometheus_multihost__inst2_1  Creating
[21:25:17.290]    Container tests_prometheus_multihost__controller_1  Created
[21:25:17.290]    Container tests_prometheus_multihost__inst1_1  Created
[21:25:17.290]    Container tests_prometheus_multihost__client_1  Created
[21:25:17.290]    Container tests_prometheus_multihost__inst2_1  Created
[21:25:17.290]    Container tests_prometheus_multihost__inst1_1  Starting
[21:25:17.290]    Container tests_prometheus_multihost__inst2_1  Starting
[21:25:17.290]    Container tests_prometheus_multihost__client_1  Starting
[21:25:17.290]    Container tests_prometheus_multihost__controller_1  Starting
[21:25:17.290]    Container tests_prometheus_multihost__inst1_1  Started
[21:25:17.290]    Container tests_prometheus_multihost__controller_1  Started
[21:25:17.290]    Container tests_prometheus_multihost__inst2_1  Started
[21:25:17.290]    Container tests_prometheus_multihost__client_1  Started
[21:25:17.290]   debconf: delaying package configuration, since apt-utils is not installed
[21:25:17.290] 
[21:25:42.883] 2 of 43 tests failed
```

I'm honestly somewhat surprised that the images wouldn't already have apt-utils installed on them by default. The failure above only happens sporadically though, not with every build. I created a branch to add `apt-utils`, but that doesn't feel like the right solution.